### DISCUSSION TITLE
deno prisma generated not work

### DISCUSSION THREAD

**Discussion by bobwatcherx** [prisma/prisma#16253] [Q&A]
i have problem after follow you tutorial
https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-deno-deploy
I have a problem when I want to generate a prism schema

my schema file
```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["deno"]
  output          = "../generated/client"
}
datasource db {
  provider = "sqlite"
  url      = "file:/prisma/sekolah.db"
}

model Sekolah{
  id Int @id @default(autoincrement())
  nama String
  kelas Int
}
```
command
`[root@fedora prismadeno]# deno run -A --unstable npm:prisma generate --data-proxy`

error message
```
[root@fedora prismadeno]# deno run -A --unstable npm:prisma generate --data-proxy
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
napi_add_env_cleanup_hook is currently not supported

up to date, audited 3 packages in 3s

found 0 vulnerabilities
Error: Could not resolve @prisma/client despite the installation that we just tried.
Please try to install it by hand with npm install @prisma/client and rerun prisma generate 🙏.
```

**deno version**
[root@fedora prismadeno]# deno --version
deno 1.28.0 (release, x86_64-unknown-linux-gnu)
v8 10.9.194.1
typescript 4.8.3

package json file
```
{
  "name": "my-prisma-project",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@prisma/client": "^4.6.1"
  }
}
```
I've installed @prisma/client and the result is still the same
thanks

**Comment by nurul3101** [comment]
Hello @bobwatcherx 👋 

Can you check if you get the same error when using the Postgres database as described in the blog?
I tried to follow the guide with the Postgres database, and it worked as expected. 

<img width="772" alt="Screenshot 2022-11-16 at 7 20 29 PM" src="https://user-images.githubusercontent.com/13049676/202198103-f690c19b-4752-4735-a9b7-c0cee5d7b0de.png">

**Reply by vladi-stevanovic** [reply]
👋 Hi @bobwatcherx, did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by bobwatcherx** [comment]
can prisma on deno use sqlite database and not postgresql .

**Reply by pantharshit00** [reply]
Hello @bobwatcherx,

Are you getting the same error which you reported when you are trying to use PostgreSQL?

**Reply by bobwatcherx** [reply]
> Hello @bobwatcherx,
> 
> Are you getting the same error which you reported when you are trying to use PostgreSQL?

for this problem , i want to try it on sqlite database . I haven't tried postgresql yet, maybe I'll try later

**Comment by yarabramasta** [comment]
I guess it's required to have node/npm installed on your machine.

I tried it before using my own machine and docker container. It works smoothly on my machine but no luck inside docker.

**Reply by yarabramasta** [reply]
I also tried using official node image as "bridge" and then copy the generated node_modules. It does generate the files, but I can't see output folder that I define in schema.prisma

It does work using npx on my own machine, but again if I use it inside docker build, it doesn't work.

**Reply by nurul3101** [reply]
Do you mind creating a [Bug Report](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) for this with a minimal reproduction so that our engineering team can investiagate?

**Comment by DimitryDushkin** [comment]
Hey everyone! Same issue here. Deno and Prisma with SQLite is not working. I also tried exclude usage .env from equation, but result is the same.  I assume that's because "deno" folder in generate client has only `edge.ts` which is not a good fit for local development.


Some input:
Schema:
```
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["deno"]
  output          = "../src/generated/prisma-client"
}

datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}
```
.env:
```
DATABASE_URL="file:./db.sqlite"
```

Code:
```
import { PrismaClient } from "../generated/prisma-client/deno/edge.ts";
```
Error:
```
Error validating datasource `db`: the URL must start with the protocol `prisma://`
```

**Reply by sebasvivas1** [reply]
Im having the same issue with a postgre database. I dont understand if its 100% necessary to use Accelerate to be able to use Prisma with Deno.

**Reply by janpio** [reply]
Right now Prisma only _officially_ supports Deno Deploy, with Prisma Accelerate: https://www.prisma.io/docs/orm/prisma-client/deployment/edge/deploy-to-deno-deploy

Support for Deno is work in progress: https://github.com/prisma/prisma/issues/2452
Currently we are blocked by a Deno details, but progress is being made to unblock: https://github.com/prisma/prisma/issues/2452#issuecomment-2160691711
(It worked for a while decently in older Deno versions, but then unfortunately broke)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
